src/rust/src/orchard_ffi.rs

```diff
@@ -74,6 +74,39 @@ pub extern "C" fn orchard_bundle_serialize(
     }
 }
 
+/// Validates the given Orchard bundle against bundle-specific consensus rules.
+///
+/// If `bundle == nullptr`, this returns `true`.
+///
+/// ## Consensus rules
+///
+/// [§4.6](https://zips.z.cash/protocol/protocol.pdf#actiondesc):
+/// - Canonical element encodings are enforced by [`orchard_bundle_parse`].
+/// - SpendAuthSig^Orchard validity is enforced by [`BatchValidator`] via
+///   [`orchard_batch_validate`].
+/// - Proof validity is enforced here.
+///
+/// [§7.1](https://zips.z.cash/protocol/protocol.pdf#txnencodingandconsensus):
+/// - `bindingSigOrchard` validity is enforced by [`BatchValidator`] via
+///   [`orchard_batch_validate`].
+#[no_mangle]
+pub extern "C" fn orchard_bundle_validate(bundle: *const Bundle<Authorized, Amount>) -> bool {
+    if let Some(bundle) = unsafe { bundle.as_ref() } {
+        let vk = unsafe { crate::ORCHARD_VK.as_ref() }.unwrap();
+
+        if bundle.verify_proof(vk).is_err() {
+            error!("Invalid Orchard proof");
```

Comment:
Is there some kind of identifier (ideally the txid) we can log here, or is that not accessible from the `Bundle`?

Reply:
It's not accessible from the bundle on its own. We could log the bundle's effecting data commitment, but if we hit this then we'll immediately see the `CheckTransaction()` error logged afterwards, so it should be clear from surrounding logic which block this corresponds to.

What I _actually_ want to do at some point is leverage `tracing`'s Spans to annotate log lines with this information. We're in a context here of checking an individual transaction; we can add that span to every logged line implicitly instead of needing to pass it through. I'm not sure how verbose that would end up becoming though.